src/software/simulated_tests/ball_moves_in_direction.py

```diff
@@ -27,20 +28,32 @@ def get_validation_status(self, world) -> ValidationStatus:
         :returns: FAILING if ball doesn't move in the direction
                   PASSING if ball moves in the direction
         """
-        validation_status = ValidationStatus.FAILING
         current_ball_position = tbots.createPoint(
             world.ball.current_state.global_position
-        )
+        ).x()
+
+        # if max displacement is not set or current ball is moving in the right direction
+        # set it and return PASSING
+        if self.max_displacement_so_far is None or (
+            (self.direction and current_ball_position > self.max_displacement_so_far)
+            or (
+                not self.direction
+                and current_ball_position < self.max_displacement_so_far
+            )
+        ):
```

Comment:
nit: the logic can be simplified. Either this, or failing
```suggestion
        if self.max_displacement_so_far is None or (
            (self.direction and current_ball_position > self.max_displacement_so_far - 0.1)
            or (
                not self.direction
                and current_ball_position < self.max_displacement_so_far + 0.1
            )
        ):
```

Reply:
Though instead of 0.1, we should have a constant (ideally passed as an arg to the constructor with a default value)